src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/HrefHandler.cs

```diff
@@ -0,0 +1,118 @@
+                }
+
+                if (currentObj == null)
+                {
+                    return null;
+                }
+
+                var val = currentObj as string;
+                if (val != null)
+                {
+                    var updated = GetHrefFromRoot(val, context);
+                    if (currentPropertyInfo != null)
+                    {
+                        ReflectionHelper.SetPropertyValue(declaringObject, currentPropertyInfo, updated);
+                    }
+                    return updated;
+                }
+
+                var list = currentObj as IList<string>;
+                if (list != null)
+                {
+                    for (var i = 0; i < list.Count; i++)
+                    {
+                        var item = list[i];
+                        if (item != null)
+                        {
+                            list[i] = GetHrefFromRoot(item, context);
+                        }
+                    }
+                    return list;
+                }
+
+                throw new NotSupportedException($"Type {currentObj.GetType()} is NOT a supported type for {nameof(HrefAttribute)}");
+            }
+
+            protected override IEnumerable<PropInfo> GetProps(Type type)
+            {
+                return from prop in base.GetProps(type)
+                       where prop.Prop.GetSetMethod() != null
+                       where !prop.Prop.IsDefined(typeof(HrefIgnoreAttribute), false)
+                       select prop;
+            }
+
+            private string GetHrefFromRoot(string originalHref, HandleModelAttributesContext context)
+            {
+                if (context.FileAndType == null || string.IsNullOrEmpty(originalHref) || !RelativePath.IsRelativePath(originalHref))
+                {
+                    return originalHref;
+                }
+
+                var result = originalHref;
+                var ft = context.FileAndType;
+                var path = (RelativePath)ft.File + (RelativePath)UriUtility.GetPath(originalHref);
+                var file = path.GetPathFromWorkingFolder().UrlDecode();
+                if (context.Host.SourceFiles.ContainsKey(file))
+                {
+                    result = file.UrlEncode().ToString() + UriUtility.GetQueryStringAndFragment(originalHref);
+                }
+
+                List<LinkSourceInfo> sources;
+                if (!context.FileLinkSources.TryGetValue(file, out sources))
+                {
+                    sources = new List<LinkSourceInfo>();
+                    context.FileLinkSources[file] = sources;
+                }
+                sources.Add(new LinkSourceInfo
+                {
+                    Target = file,
+                    Anchor = UriUtility.GetFragment(originalHref),
+                    SourceFile = ft.File,
+                });
+                context.LinkToFiles.Add(file);
```

Comment:
Seems it also makes sense to add

---
In reply to: [111533603](https://github.com/dotnet/docfx/pull/1543#discussion_r111533603) [](ancestors = 111533603)